spec_util/ir_hash/wrappers.go

```diff
@@ -0,0 +1,17 @@
+package ir_hash
+
+import (
+	"encoding/base64"
+
+	pb "github.com/akitasoftware/akita-ir/go/api_spec"
+)
+
+func HashWitnessToString(w *pb.Witness) string {
```

Comment:
Hashing can't fail?  What would happen if I changed the IR but didn't change this file in the future, and then tried to hash a witness with added/removed fields?

Reply:
Because it's not using reflection, there is no way to know whether additional fields are present or not, so it would silently exclude them from the hash.

It would be nice if we could attach some version number to the generated go objects (that could be easily checked)--- let me see if we can do that with fileDescriptor.

Reply:
https://github.com/akitasoftware/akita-libs/pull/71/commits/0d71d81eb46c8c9108929d300700909e915b9778 stores hashes of the protobuf "file descriptors", which should pick up any added types and cause a panic during tests.